src/Microsoft.NetCore.Analyzers/Core/Performance/DoNotUseCountWhenAnyCanBeUsed.Fixer.cs

```diff
@@ -53,27 +53,50 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
             var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
             var node = root.FindNode(context.Span);
-            var isAsync = context.Diagnostics[0].Id == DoNotUseCountWhenAnyCanBeUsedAnalyzer.AsyncRuleId;
+            var properties = context.Diagnostics[0].Properties;
+            var shouldNegateKey = properties.ContainsKey(DoNotUseCountWhenAnyCanBeUsedAnalyzer.ShouldNegateKey);
+            var isAsync = properties.ContainsKey(DoNotUseCountWhenAnyCanBeUsedAnalyzer.IsAsyncKey) ||
+                context.Diagnostics[0].Id == DoNotUseCountWhenAnyCanBeUsedAnalyzer.AsyncRuleId;
 
             if (node is object &&
-                this.TryGetFixer(node, isAsync, out var expression, out var arguments, out var negate))
+                properties.TryGetValue(DoNotUseCountWhenAnyCanBeUsedAnalyzer.OperationKey, out var operation) &&
+                this.TryGetFixer(node, operation, isAsync, out var expression, out var arguments))
             {
                 context.RegisterCodeFix(
-                    new DoNotUseCountWhenAnyCanBeUsedCodeAction(isAsync, context.Document, node, expression, arguments, negate),
+                    new DoNotUseCountWhenAnyCanBeUsedCodeAction(isAsync, context.Document, node, expression, arguments, shouldNegateKey),
                     context.Diagnostics);
             }
         }
 
+
+        /// <summary>
+        /// Determines whether the specified <paramref name="value"/> is a match for <paramref name="expected"/>.
+        /// </summary>
+        /// <param name="value">The value to match.</param>
+        /// <param name="expected">The expected value.</param>
+        /// <returns><see langword="true" /> if the specified <paramref name="value"/> is a match for <paramref name="expected"/>; otherwise, <see langword="false" />.</returns>
+        protected static bool IsMatch(object value, int expected)
```

Comment:
We have some `TryConvertToUInt64` helper methods in DiagnosticHelpers. Can those be used to simplify this?

Reply:
This is note used. Forgot to remove it.